packages/server/src/provider/in-memory-provider/in-memory-provider.ts

```diff
@@ -0,0 +1,152 @@
+ */
+export class InMemoryProvider implements Provider {
+  public readonly events = new OpenFeatureEventEmitter();
+  public readonly runsOn = 'server';
+  readonly metadata = {
+    name: 'In-Memory Provider',
+  } as const;
+  private _flagConfiguration: FlagConfiguration;
+
+  constructor(flagConfiguration: FlagConfiguration = {}) {
+    this._flagConfiguration = { ...flagConfiguration };
+  }
+
+  /**
+   * Overwrites the configured flags.
+   * @param { FlagConfiguration } flagConfiguration new flag configuration
+   */
+  putConfiguration(flagConfiguration: FlagConfiguration) {
+    const flagsChanged = Object.entries(flagConfiguration)
+      .filter(([key, value]) => this._flagConfiguration[key] !== value)
+      .map(([key]) => key);
+
+    this._flagConfiguration = { ...flagConfiguration };
+    this.events.emit(ProviderEvents.ConfigurationChanged, { flagsChanged });
+  }
+
+  resolveBooleanEvaluation(
+    flagKey: string,
+    defaultValue: boolean,
+    context?: EvaluationContext,
+    logger?: Logger,
+  ): Promise<ResolutionDetails<boolean>> {
+    return Promise.resolve(this.resolveFlagWithReason<boolean>(flagKey, defaultValue, context, logger));
+  }
+
+  resolveNumberEvaluation(
+    flagKey: string,
+    defaultValue: number,
+    context?: EvaluationContext,
+    logger?: Logger,
+  ): Promise<ResolutionDetails<number>> {
+    return Promise.resolve(this.resolveFlagWithReason<number>(flagKey, defaultValue, context, logger));
+  }
+
+  async resolveStringEvaluation(
+    flagKey: string,
+    defaultValue: string,
+    context?: EvaluationContext,
+    logger?: Logger,
+  ): Promise<ResolutionDetails<string>> {
+    return Promise.resolve(this.resolveFlagWithReason<string>(flagKey, defaultValue, context, logger));
+  }
+
+  async resolveObjectEvaluation<T extends JsonValue>(
+    flagKey: string,
+    defaultValue: T,
+    context?: EvaluationContext,
+    logger?: Logger,
+  ): Promise<ResolutionDetails<T>> {
+    return Promise.resolve(this.resolveFlagWithReason<T>(flagKey, defaultValue, context, logger));
+  }
+
+  private resolveFlagWithReason<T extends JsonValue | FlagValueType>(
+    flagKey: string,
+    defaultValue: T,
+    ctx?: EvaluationContext,
+    logger?: Logger,
+  ): ResolutionDetails<T> {
+    try {
+      const resolutionResult = this.lookupFlagValue(flagKey, defaultValue, ctx, logger);
+
+      if (typeof resolutionResult?.value != typeof defaultValue) {
+        throw new TypeMismatchError();
+      }
+
+      return resolutionResult;
+    } catch (error: unknown) {
+      let errorCode;
+      switch (true) {
+        case error instanceof TypeMismatchError:
+          errorCode = ErrorCode.TYPE_MISMATCH;
+          break;
+        case error instanceof FlagNotFoundError:
+          errorCode = ErrorCode.FLAG_NOT_FOUND;
+          break;
+        case error instanceof GeneralError:
+          errorCode = ErrorCode.PARSE_ERROR;
```

Comment:
Why are you mapping a general error to a parse error?

Reply:
I didn't see this, I'll look into it.

Reply:
I did it based on [this comment](https://github.com/open-feature/js-sdk-contrib/pull/607#discussion_r1350716854).
I'm not really happy with it. Since it's rather difficult to make sure that an existing variant will always be available in a context based evaluation a `VariantNotFound` error would fit better in this case.